server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeMetaWriter.java

```diff
@@ -0,0 +1,226 @@
+        metaOut.writeLong(MAGIC_MARKER);
+
+        // version
+        metaOut.writeInt(VERSION);
+
+        // star tree field name
+        metaOut.writeString(starTreeFieldName);
+
+        // star tree field type
+        metaOut.writeString(compositeFieldType.getName());
+    }
+
+    /**
+     * Writes the star-tree metadata.
+     *
+     * @param metaOut                   the IndexOutput to write the metadata
+     * @param writeState                the segment write state
+     * @param metricAggregatorInfos     the list of metric aggregator information
+     * @param starTreeField             the star tree field
+     * @param segmentAggregatedDocCount the aggregated document count for the segment
+     * @param dataFilePointer           the file pointer to the start of the star-tree data
+     * @param dataFileLength            the length of the star-tree data file
+     * @throws IOException if an I/O error occurs while writing the metadata
+     */
+    private static void writeMeta(
+        IndexOutput metaOut,
+        SegmentWriteState writeState,
+        List<MetricAggregatorInfo> metricAggregatorInfos,
+        StarTreeField starTreeField,
+        Integer segmentAggregatedDocCount,
+        long dataFilePointer,
+        long dataFileLength
+    ) throws IOException {
+
+        // number of dimensions
+        metaOut.writeInt(starTreeField.getDimensionsOrder().size());
+
+        // dimensions
+        for (Dimension dimension : starTreeField.getDimensionsOrder()) {
+            int dimensionFieldNumber = writeState.fieldInfos.fieldInfo(dimension.getField()).getFieldNumber();
+            metaOut.writeInt(dimensionFieldNumber);
+        }
+
+        // number of metrics
+        metaOut.writeInt(metricAggregatorInfos.size());
+
+        // metric - metric stat pair
+        for (MetricAggregatorInfo metricAggregatorInfo : metricAggregatorInfos) {
+            String metricName = metricAggregatorInfo.getField();
+            String metricStatName = metricAggregatorInfo.getMetricStat().getTypeName();
+            metaOut.writeString(metricName);
+            metaOut.writeString(metricStatName);
+        }
+
+        // segment aggregated document count
+        metaOut.writeInt(segmentAggregatedDocCount);
+
+        // max leaf docs
+        metaOut.writeInt(starTreeField.getStarTreeConfig().maxLeafDocs());
+
+        // number of skip star node creation dimensions
+        metaOut.writeInt(starTreeField.getStarTreeConfig().getSkipStarNodeCreationInDims().size());
+
+        // skip star node creations
+        for (String dimension : starTreeField.getStarTreeConfig().getSkipStarNodeCreationInDims()) {
+            int dimensionFieldNumber = writeState.fieldInfos.fieldInfo(dimension).getFieldNumber();
+            metaOut.writeInt(dimensionFieldNumber);
+        }
+
+        // star tree build-mode
+        metaOut.writeString(starTreeField.getStarTreeConfig().getBuildMode().getTypeName());
```

Comment:
We should be able to use enum ordinals or constants to identify the associated types
For example : 
```
  // indicates docvalues type
  static final byte NUMERIC = 0;
  static final byte BINARY = 1;
  static final byte SORTED = 2;
  static final byte SORTED_SET = 3;
  static final byte SORTED_NUMERIC = 4;
```